### ISSUE TITLE
updateOnEventListener being called before this.player is asynchronously set

### ISSUE THREAD

**Issue by ness-EE** [jwplayer/jwplayer-react#17]
Hi,

I have an implementation that is failing if there are any `on` callbacks like `onPlay`.

```
TypeError: Cannot read properties of null (reading 'off')
    at f.updateOnEventListener (http://localhost:6006/vendors-node_modules_date-io_dayjs_build_index_esm_js-node_modules_jwplayer_jwplayer-react_li-bbaa53.iframe.bundle.js:429:11411)
    at f.shouldComponentUpdate (http://localhost:6006/vendors-node_modules_date-io_dayjs_build_index_esm_js-node_modules_jwplayer_jwplayer-react_li-bbaa53.iframe.bundle.js:429:10598)
    at checkShouldComponentUpdate (http://localhost:6006/vendors-node_modules_date-io_dayjs_build_index_esm_js-node_modules_jwplayer_jwplayer-react_li-bbaa53.iframe.bundle.js:168985:33)
    at updateClassInstance (http://localhost:6006/vendors-node_modules_date-io_dayjs_build_index_esm_js-node_modules_jwplayer_jwplayer-react_li-bbaa53.iframe.bundle.js:169499:62)
    at updateClassComponent (http://localhost:6006/vendors-node_modules_date-io_dayjs_build_index_esm_js-node_modules_jwplayer_jwplayer-react_li-bbaa53.iframe.bundle.js:173892:20)
    at beginWork (http://localhost:6006/vendors-node_modules_date-io_dayjs_build_index_esm_js-node_modules_jwplayer_jwplayer-react_li-bbaa53.iframe.bundle.js:175533:16)
    at HTMLUnknownElement.callCallback (http://localhost:6006/vendors-node_modules_date-io_dayjs_build_index_esm_js-node_modules_jwplayer_jwplayer-react_li-bbaa53.iframe.bundle.js:160405:14)
    at Object.invokeGuardedCallbackDev (http://localhost:6006/vendors-node_modules_date-io_dayjs_build_index_esm_js-node_modules_jwplayer_jwplayer-react_li-bbaa53.iframe.bundle.js:160454:16)
    at invokeGuardedCallback (http://localhost:6006/vendors-node_modules_date-io_dayjs_build_index_esm_js-node_modules_jwplayer_jwplayer-react_li-bbaa53.iframe.bundle.js:160516:31)
    at beginWork$1 (http://localhost:6006/vendors-node_modules_date-io_dayjs_build_index_esm_js-node_modules_jwplayer_jwplayer-react_li-bbaa53.iframe.bundle.js:180419:7)
```

Looking at the package it seems that the player is not being set by the time that `updateOnEventListener` is called and there is no check in `updateOnEventListener` that `this.player` is not null

just for reference here is my implementation:

```
<div className={`jw-player ${className || ''}`.trim()}>
      <JWPlayerReact
        library={playerScriptUrl}
        file={signedUrl}
        autostart={autoPlay}
        didMountCallback={internalOnReady}
        onPlay={debounce(onEvent('play'), 500)}
        onAutoStart={debounce(onEvent('play'), 500)}
        onPause={onEvent('pause')}
        onResume={onEvent('resume')}
        onError={onEvent('error')}
        mediaId={mediaId}
        advertising={config.advertising}
      />
    </div>
```

I'm part way through the implementation, so not sure if things like `onError` are valid yet. Removing all of them except `onPlay` produces the same error. Just for a bit of context I'm migrating from `react-jw-player` as I need the player object from `didMountCallback`

**Comment by ness-EE** [comment]
I've taken 5 minutes to check out the repo and add it locally to my project.

I'm afraid my findings are exactly as I stated in this issue

If I go through this file https://github.com/jwplayer/jwplayer-react/blob/main/src/jwplayer.jsx I can follow how the problem occurs.

`this.player = null;`

is set before component mount

`this.player = await this.createPlayer();`

as you can see this is now set through an asynchronous request so we have no idea how long it will take.

Whilst that is happening there may be prop and state changes, invoking `shouldComponentUpdate`

This method has a terrible side effect in it which should be moved - can the comparison be done with prevProps using `componentDidUpdate`?

`this.state.playerLoadPromise` resolves, invoking `shouldComponentUpdate`, which then invokes the side effect. This calls `updateOnEventListener` before `this.player` is set - and there you have your race condition, which is why you can't reproduce it.

Also as you're mutating `this.player` should it not be in the state?

**Comment by mamaddox** [comment]
Hi @ness-EE, this should be fixed in 1.1.1.
